nixos/modules/config/shells-environment.nix

```diff
@@ -184,8 +192,10 @@ in
 
         ${cfg.extraInit}
 
-        # ~/bin if it exists overrides other bin directories.
-        export PATH="$HOME/bin:$PATH"
+        ${if cfg.homeBinInPath then ''
```

Comment:
What you want to use here is `optionalString cfg.homeBinInPath` then you do not need the weird empty string in the else clause.

Reply:
Done.  Thanks.